src/views/components/item.jsx

```diff
@@ -1,16 +1,56 @@
 import React from 'react';
-
+import ItemStore from '../../store/item-store.js';
+import { Link } from 'react-router';
 export default class Item extends React.Component {
+    constructor(props) {
+        super(props);
+        if (props.item !== null)
+            this.state = {item: props.item};
+        else
+            this.state = {item: null};
+    }
 
-    render() {
-        if (this.props.item === null) {
-            return <i>No item found</i>;
+    componentDidMount() {
+        if (this.state.item === null) {
+            ItemStore.searchItemByAddress(this.props.params.address).then(() => {
+                this.setState({
+                    item: ItemStore.getItem()
+                })
+            });
         }
+    }
+
+    render() {
+        if (this.state.item === null)
+            return (<i>Item is loading...</i>);
```

Comment:
Returning a loading message like this while the item is null is a good way to do this, though :+1: